lib/internal/util.js

```diff
@@ -184,17 +183,12 @@ function emitExperimentalWarning(feature) {
 }
 
 function filterDuplicateStrings(items, low) {
-  const map = new Map();
+  const set = new Set();
   for (let i = 0; i < items.length; i++) {
-    const item = items[i];
-    const key = item.toLowerCase();
-    if (low) {
-      map.set(key, key);
-    } else {
-      map.set(key, item);
-    }
+    const val = low ? items[i].toLowerCase() : items[i];
+    set.add(val);
   }
-  return ArrayFrom(map.values()).sort();
+  return ArrayFrom(set.values()).sort();
 }
```

Comment:
You can simplify this even further, like this:
```js
function filterDuplicateStrings(items, low) {
  if (low) items = items.map((s) => s.toLowerCase());
  return ArrayFrom(new Set(items)).sort();
}
```
However! Neither algorithm is equivalent to the original. When `low === false`, strings with different casings should map to a single string.

Example: `filterDuplicateStrings(['one', 'ONE', 'onE'])` - that should return `['one']` but returns `['one', 'ONE', 'onE']` with your (or my) change.

Reply:
Hmm, make sense, but in your example, it is giving preference to the only first string and what is cause of creating such function because neither of the tests broke with this change. Just curious to know.

Reply:
@BridgeAR Are you referring to something around these lines?

```diff
diff --git a/lib/internal/crypto/util.js b/lib/internal/crypto/util.js
index 286efc39a6..48e9d79707 100644
--- a/lib/internal/crypto/util.js
+++ b/lib/internal/crypto/util.js
@@ -58,9 +58,12 @@ function toBuf(val, encoding) {
   return val;
 }
 
-const getCiphers = cachedResult(() => filterDuplicateStrings(_getCiphers()));
-const getHashes = cachedResult(() => filterDuplicateStrings(_getHashes()));
-const getCurves = cachedResult(() => filterDuplicateStrings(_getCurves()));
+const getCiphers = cachedResult(() =>
+  filterDuplicateStrings(_getCiphers().sort()));
+const getHashes = cachedResult(() =>
+  filterDuplicateStrings(_getHashes().sort()));
+const getCurves = cachedResult(() =>
+  filterDuplicateStrings(_getCurves().sort()));
 
 function setEngine(id, flags) {
   validateString(id, 'id');
diff --git a/lib/internal/util.js b/lib/internal/util.js
index 2737d78c6b..b6ba68c6e3 100644
--- a/lib/internal/util.js
+++ b/lib/internal/util.js
@@ -186,7 +186,7 @@ function filterDuplicateStrings(items, low) {
   const nonUniqArr = low ?
     items.map((item) => item.toLowerCase()) :
     items;
-  return ArrayFrom(new Set(nonUniqArr)).sort();
+  return ArrayFrom(new Set(nonUniqArr));
 }
 
 function cachedResult(fn) {
diff --git a/lib/tls.js b/lib/tls.js
index 2ccbe409c9..524be105bb 100644
--- a/lib/tls.js
+++ b/lib/tls.js
@@ -80,7 +80,7 @@ else
 
 
 exports.getCiphers = internalUtil.cachedResult(
-  () => internalUtil.filterDuplicateStrings(getSSLCiphers(), true)
+  () => internalUtil.filterDuplicateStrings(getSSLCiphers().sort(), true)
 );
 
 let rootCertificates;
```